examples/organization_trail_with_external_bucket/main.tf

```diff
@@ -0,0 +1,81 @@
+locals {
+  name            = "org-trail-bucket-example"
+  account_id      = data.aws_caller_identity.current.account_id
+  region          = data.aws_region.current.name
+  organization_id = data.aws_organizations_organization.current.id
+}
+
+resource "random_pet" "main" {
+  length = 2
+}
+
+#####################################################################
+### Bucket with policy to ensure it has permissions for cloudtrail
+#####################################################################
+resource "aws_s3_bucket" "cloudtrail" {
+  bucket        = "${local.name}-${random_pet.main.id}"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_policy" "cloudtrail" {
+  bucket = aws_s3_bucket.cloudtrail.bucket
+  policy = data.aws_iam_policy_document.org_s3.json
```

Comment:
We'll first need to upgrade our S3 module because when doing `terraform init` both modules are downloaded but in this case they use different versions hence not compatible.